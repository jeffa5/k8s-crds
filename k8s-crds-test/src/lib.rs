#[path = "generated/generated.rs"]
pub mod generated;

#[cfg(test)]
mod tests {
    use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;

    use super::*;

    #[test]
    fn serde_basic() {
        use generated::basic_object::stable_example_com::v1::cron_tab::*;

        let ct = CronTab {
            metadata: ObjectMeta {
                name: Some("test".into()),
                ..Default::default()
            },
            spec: Spec {
                cron_spec: "* * * * *".into(),
                image: "nginx".into(),
                replicas: 2,
            },
        };

        let s = serde_yaml::to_string(&ct).unwrap();
        insta::assert_snapshot!(s);

        let ct_d = serde_yaml::from_str(&s).unwrap();

        assert_eq!(ct, ct_d);
    }

    #[test]
    fn serde_double() {
        use generated::double_object::stable_example_com::v1::cron_tab::*;

        let ct = CronTab {
            metadata: ObjectMeta {
                name: Some("test".into()),
                ..Default::default()
            },
            obj1: Obj1 {
                a: Obj1A { b: "BB".into() },
            },
            obj2: Obj2 { a: Obj2A { c: 4 } },
        };

        let s = serde_yaml::to_string(&ct).unwrap();

        insta::assert_snapshot!(s);

        let ct_d = serde_yaml::from_str(&s).unwrap();

        assert_eq!(ct, ct_d);
    }

    #[test]
    fn serde_long() {
        use generated::long_names::stable_example_com::v1::cron_tab::*;

        let ct = CronTab {
            metadata: ObjectMeta {
                name: Some("test".into()),
                ..Default::default()
            },
            a: A {
                b: B {
                    c: C {
                        d: D { e: 2 },
                        f: F { g: "gg".into() },
                    },
                },
            },
        };

        let s = serde_yaml::to_string(&ct).unwrap();

        insta::assert_snapshot!(s);

        let ct_d = serde_yaml::from_str(&s).unwrap();

        assert_eq!(ct, ct_d);
    }
}
