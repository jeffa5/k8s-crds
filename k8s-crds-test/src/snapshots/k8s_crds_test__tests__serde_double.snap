---
source: k8s-crds-test/src/lib.rs
assertion_line: 48
expression: s
---
---
metadata:
  name: test
obj1:
  a:
    b: BB
obj2:
  a:
    c: 4

